blog/2023-07-31-new-pre-post-command-plugin-hooks.mdx

```diff
@@ -0,0 +1,165 @@
+  these in this post.
+authors: [travishathaway]
+tags: [conda, plugins]
+image: img/blog/2023-07-31-new-pre-post-command-plugin-hooks/pre-post-command-image.jpg
+---
+
+[conda-issue]: https://github.com/conda/conda/issues/12138
+[conda-guard]: https://github.com/travishathaway/conda-guard
+[conda-plugin-docs]: https://docs.conda.io/projects/conda/en/stable/dev-guide/plugins/index.html
+[matrix]: https://matrix.to/#/#conda:matrix.org
+
+import Image from '@theme/IdealImage';
+import banner from '@site/static/img/blog/2023-07-31-new-pre-post-command-plugin-hooks/pre-post-command-image.jpg';
+
+<Image img={banner} alt="Banner image for blog post; New pre and post command plugin hooks - picture of a plugin outlet"/>
+
+<p>
+    <i>
+        <small>
+            Image credit: <a href="https://unsplash.com/photos/7d8pxcMVl7A">Kier in Sight Archives on Unsplash</a>
+        </small>
+    </i>
+</p>
+
+With the latest conda release (`23.7.0` at the time this post was written), the ability to define
+two new plugin hooks was introduced: "pre command" and "post command". These two new plugin hooks give
+plugin authors the ability to execute code before and after conda commands are run. In this blog post,
+we provide more details on how and why you may use these to extend the default behavior of conda.
+
+<!-- truncate -->
+
+:::info
+
+To check out a fully functional example, check out the [conda-guard][conda-guard] project
+to see how these new plugin hooks are used in practice.
+:::
+
+
+---
+
+To explain how these plugin hooks can be used, we will cover two examples.
+
+## Conda guard and the "pre command" hook
+
+The project linked to above is called [`conda-guard`](https://github.com/travishathaway/conda-guard), and it extends conda by adding functionality to "guard"
+a conda environment from being changed. Sometimes users may want to do this to protect themselves from
+[unintentionally modifying their base environments][conda-issue].
+
+Using the "pre command" plugin hook, we can first specify which commands we want it to run for and then
+run a function that executes before the command gets executed. By doing this, we can figure out whether
+the environment is guarded or not and exit early if so. Below is how such a plugin hook would be defined:
+
+```python
+from conda.plugins import hookimpl, CondaPreCommand
+
+def conda_guard_pre_commands_action(command: str) -> None:
+    """Checks to see if the current environment is guarded"""
+    current_environment = get_current_environment()
+    
+    if is_guarded(current_environment):
+        raise CondaError(
+            f"Current environment is guarded. Run `conda guard {environment}` to "
+            "remove guard."
+        )
+
+@hookimpl
+def conda_pre_commands():
+    yield CondaPreCommand(
+        name=f"conda_guard_pre_command",
+        action=conda_guard_pre_commands_action,
+        run_for={"install", "remove", "update", "env_update", "env_remove"},
+    )
+```
```

Comment:
```suggestion
@plugins.hookimpl
def conda_pre_commands():
    yield plugins.CondaPreCommand(
        name=f"conda_guard_pre_command",
        action=conda_guard_pre_commands_action,
        run_for={"install", "remove", "update", "env_update", "env_remove"},
    )
```

I think this makes it a little more readable since `hookimpl` is rather obtuse